src/map/script.cpp

```diff
@@ -6687,15 +6687,22 @@ BUILDIN_FUNC(viewpoint)
 	int type,x,y,id,color;
 	TBL_PC* sd;
 
+	if (script_hasdata(st, 7)) {
+		script_charid2sd(7, sd)
+	else
+		script_rid2sd(sd))
+
+	if(!sd){
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
```

Comment:
After re-reading the check just need to be
```

	if (!script_charid2sd(7, sd)) {
		st->state = END;
		return SCRIPT_CMD_FAILURE;
	}
```

Reply:
MVP